app/models/export/activity_forecast_columns.rb

```diff
@@ -0,0 +1,61 @@
+class Export::ActivityForecastColumns
+  def initialize(activities:)
+    @activities = activities
+  end
+
+  def headers
+    return [] if @activities.empty?
+
+    forecast_financial_quarter_range.map do |financial_quarter|
+      "Forecast #{financial_quarter}"
+    end
+  end
+
+  def rows
+    return [] if @activities.empty?
+
+    @activities.map { |activity|
+      [activity.id, forecast_data(activity)]
+    }.to_h
+  end
+
+  private
+
+  def forecast_data(activity)
+    forecast_financial_quarter_range.map { |fq|
+      forecasts_to_hash.fetch([activity.id, fq.quarter, fq.financial_year.start_year], 0)
+    }
+  end
+
+  def activity_ids
+    @activities.pluck(:id)
+  end
+
+  def forecasts
+    overview = ForecastOverview.new(activity_ids)
+    @_forecasts ||= overview.latest_values
+  end
+
+  def forecasts_to_hash
+    @_forecasts_to_hash ||= forecasts.each_with_object({}) { |forecast, hash|
+      hash[[forecast.parent_activity_id, forecast.financial_quarter, forecast.financial_year]] = forecast.value
+    }
+  end
+
+  def all_financial_quarters_with_forecasts
+    return [] unless forecasts.present?
+    forecasts.map(&:own_financial_quarter).uniq
+  end
+
+  def forecast_financial_quarter_range
+    @_forecast_quarter_range ||= begin
+      return [] if all_financial_quarters_with_forecasts.blank?
+
+      Range.new(all_financial_quarters_with_forecasts.min, all_financial_quarters_with_forecasts.max)
+    end
+  end
+
+  def financial_quarters
+    all_financial_quarters_with_forecasts
+  end
```

Comment:
I don't think this method is being used?

Maybe you were planning to call this generally rather than `all_financial_quarters_with_forecasts`?

Reply:
It is just a hangover from the spending breakdown report where it would have included 'financial quarters with acutals' in addition - it does nothing now, I'll remove it - thank you!